beacon_node/http_api/src/lib.rs

```diff
@@ -1125,6 +1125,81 @@ pub fn serve<T: BeaconChainTypes>(
             },
         );
 
+    // GET beacon/states/{state_id}/pending_deposits
+    let get_beacon_state_pending_deposits = beacon_states_path
+        .clone()
+        .and(warp::path("pending_deposits"))
+        .and(warp::path::end())
+        .then(
+            |state_id: StateId,
+             task_spawner: TaskSpawner<T::EthSpec>,
+             chain: Arc<BeaconChain<T>>| {
+                task_spawner.blocking_json_task(Priority::P1, move || {
+                    let (data, execution_optimistic, finalized) = state_id
+                        .map_state_and_execution_optimistic_and_finalized(
+                            &chain,
+                            |state, execution_optimistic, finalized| {
+                                if !state.fork_name_unchecked().electra_enabled() {
+                                    return Err(warp_utils::reject::pre_electra_not_supported(
+                                        format!("state at epoch {} is not activated for Electra", state.current_epoch())
+                                    ));
+                                }
+
+                                let Ok(deposits) = state.pending_deposits() else {
+                                    return Err(warp_utils::reject::custom_bad_request("Pending deposits not found".to_string()));
+                                };
+
+                                Ok((deposits.iter().cloned().collect::<Vec<_>>(), execution_optimistic, finalized))
+                            }
+                        )?;
+
+                        Ok(api_types::ExecutionOptimisticFinalizedResponse {
+                            data,
+                            execution_optimistic: Some(execution_optimistic),
+                            finalized: Some(finalized),
+                        })
+                })
+            },
+        );
+
+    // GET beacon/states/{state_id}/pending_partial_withdrawals
+    let get_beacon_state_pending_partial_withdrawals = beacon_states_path
+        .clone()
+        .and(warp::path("pending_partial_withdrawals"))
+        .and(warp::path::end())
+        .then(
+            |state_id: StateId,
+             task_spawner: TaskSpawner<T::EthSpec>,
+             chain: Arc<BeaconChain<T>>| {
+                task_spawner.blocking_json_task(Priority::P1, move || {
+                    let (data, execution_optimistic, finalized) = state_id
+                        .map_state_and_execution_optimistic_and_finalized(
+                            &chain,
+                            |state, execution_optimistic, finalized| {
+                                if !state.fork_name_unchecked().electra_enabled() {
```

Comment:
we probably don't need this check or the new error type because the case where the state doesn't have the field is handled below with `custom_bad_request`